dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/pages/datasource/DataSourcePage.java

```diff
@@ -0,0 +1,161 @@
+
+    @FindBy(className = "dataSourceItems")
+    private List<WebElement> dataSourceItemsList;
+
+    @FindBys({
+        @FindBy(className = "el-popconfirm"),
+        @FindBy(className = "el-button--primary"),
+    })
+    private List<WebElement> buttonConfirm;
+
+    private final CreateDataSourceForm createDataSourceForm;
+
+    public DataSourcePage(RemoteWebDriver driver) {
+        super(driver);
+
+        createDataSourceForm = new CreateDataSourceForm();
+    }
+
+    public DataSourcePage createDataSource(String dataSourceType, String dataSourceName, String dataSourceDescription, String ip, String port, String userName, String password, String database,
+                                           String jdbcParams) {
+        buttonCreateDataSource().click();
+
+        createDataSourceForm().btnDataSourceTypeDropdown().click();
+
+        new WebDriverWait(driver, 10).until(ExpectedConditions.visibilityOfElementLocated(new By.ById("dialogCreateDataSource")));
+
+        createDataSourceForm().selectDataSourceType()
+            .stream()
+            .filter(it -> it.getText().contains(dataSourceType.toUpperCase()))
+            .findFirst()
+            .orElseThrow(() -> new RuntimeException(String.format("No %s in data source type list", dataSourceType.toUpperCase())))
+            .click();
+
+        createDataSourceForm().inputDataSourceName().sendKeys(dataSourceName);
+        createDataSourceForm().inputDataSourceDescription().sendKeys(dataSourceDescription);
+        createDataSourceForm().inputIP().sendKeys(ip);
+        createDataSourceForm().inputPort().clear();
+        createDataSourceForm().inputPort().sendKeys(port);
+        createDataSourceForm().inputUserName().sendKeys(userName);
+        createDataSourceForm().inputPassword().sendKeys(password);
+        createDataSourceForm().inputDataBase().sendKeys(database);
+
+        if (!"".equals(jdbcParams)) {
+            createDataSourceForm().inputJdbcParams().sendKeys(jdbcParams);
+        }
+
+        createDataSourceForm().buttonSubmit().click();
+
+        return this;
+    }
+
+    public DataSourcePage delete(String name) {
+        dataSourceItemsList()
+            .stream()
+            .filter(it -> it.getText().contains(name))
+            .flatMap(it -> it.findElements(By.id("btnDelete")).stream())
+            .filter(WebElement::isDisplayed)
+            .findFirst()
+            .orElseThrow(() -> new RuntimeException("No delete button in data source list"))
+            .click();
+
+        buttonConfirm()
+            .stream()
+            .filter(WebElement::isDisplayed)
+            .findFirst()
+            .orElseThrow(() -> new RuntimeException("No confirm button when deleting"))
+            .click();
+
+        return this;
+    }
+
+    @Getter
+    public class CreateDataSourceForm {
+        CreateDataSourceForm() {
+            PageFactory.initElements(driver, this);
+        }
+
+        @FindBy(className = "OptionsDataSourceType")

```

Comment:
According to the UI team, UI class names are kebab-case, like `options-datasource-type`, UI id names are camel-case (you are right in ID naming)